tests/basic/async_dr_test.go

```diff
@@ -637,13 +639,274 @@ var _ = Describe("{StorkctlPerformFailoverFailbackeckEsClusterwide}", func() {
+				statsData := make(map[string]string)
+				statsData["numOfNodes"] = fmt.Sprintf("%d", numOfNodes)
+				statsData["fromVersion"] = currPXVersion
+				statsData["toVersion"] = updatedPXVersion
+				statsData["duration"] = fmt.Sprintf("%d mins", durationInMins)
+				statsData["status"] = upgradeStatus
+				dash.UpdateStats("px-upgrade-stats", "px-enterprise", "upgrade", majorVersion, statsData)
+			}
+		})
+
+		Step("Perform failover and failback", func() {
+			time.Sleep(2 * time.Duration(migrationInterval) * time.Minute)
+			log.InfoD("Perform failover and failback")
+			err = SetCustomKubeConfig(asyncdr.SecondCluster)
+			log.FailOnError(err, "Switching context to dest cluster failed")
+			err = Inst().S.RefreshNodeRegistry()
+			log.FailOnError(err, "Node registry refresh failed")
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "Refresh Driver end points failed")
+			extraArgsFailoverFailback := map[string]string{
+				"kubeconfig": kubeConfigPathDest,
+			}
+			failoverParam := failoverFailbackParam{
+				action:                    "failover",
+				failoverOrFailbackNs:      defaultNs,
+				migrationSchedName:        migrationSchedName,
+				configPath:                kubeConfigPathDest,
+				single:                    false,
+				skipSourceOp:              false,
+				includeNs:                 false,
+				excludeNs:                 false,
+				extraArgsFailoverFailback: extraArgsFailoverFailback,
+				contexts:                  contexts,
+			}
+
+			performFailoverFailback(failoverParam)
+
+			err = hardSetConfig(kubeConfigPathDest)
+			log.FailOnError(err, "Error setting destination config: %v", err)
+			extraArgs["kubeconfig"] = kubeConfigPathDest
+			newMigSched := migrationSchedName + "-rev"
+			createMigSchdAndValidateMigration(newMigSched, cpName, defaultNs, kubeConfigPathDest, extraArgs)
+			failback := failoverFailbackParam{
+				action:                    "failback",
+				failoverOrFailbackNs:      defaultNs,
+				migrationSchedName:        newMigSched,
+				configPath:                kubeConfigPathDest,
+				single:                    false,
+				skipSourceOp:              false,
+				includeNs:                 false,
+				excludeNs:                 false,
+				extraArgsFailoverFailback: extraArgsFailoverFailback,
+				contexts:                  contexts,
+			}
+			performFailoverFailback(failback)
+		})
+
+		err = SetCustomKubeConfig(asyncdr.FirstCluster)
+		log.FailOnError(err, "Switching context to source cluster failed")
+		err = Inst().S.RefreshNodeRegistry()
+		log.FailOnError(err, "Node registry refresh failed")
+		err = Inst().V.RefreshDriverEndpoints()
+		log.FailOnError(err, "Refresh Driver end points failed")
+
+		Step("Destroy apps", func() {
+			log.InfoD("Destroy apps")
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+			for _, ctx := range contexts {
+				TearDownContext(ctx, opts)
+			}
+		})
```

Comment:
Should we stop all migrations and then destroy apps?

Reply:
Migrations suspended both side automatically and destination apps already down. So I think cleaning up just source context is enough here. I will discuss once and show you. Thanks.